packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -1423,17 +1424,60 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     const res = resW as ResponseType<typeof stateP.controlState>;
     if (Either.isRight(res)) {
       if (!res.right.match) {
+        if (res.right.updatedHashes?.length) {
+          const typeFields = getIndexTypes(stateP);
+          const nonTypeFields = res.right.updatedHashes.filter(
+            (field) => !typeFields.includes(field)
+          );
+
+          if (nonTypeFields.length) {
+            return {
+              ...stateP,
+              controlState: 'UPDATE_TARGET_MAPPINGS_PROPERTIES',
+              updatedTypesQuery: Option.none,
+              logs: [
+                ...stateP.logs,
+                {
+                  level: 'info',
+                  message: `Kibana is performing a compatible update and some root fields have been updated. All SO documents must be updated. Updated root fields: ${nonTypeFields}.`,
+                },
+              ],
+            };
+          } else {
+            // all updated hashes correspond to SO types
+            const updatedTypes = res.right.updatedHashes;
+            return {
+              ...stateP,
+              controlState: 'UPDATE_TARGET_MAPPINGS_PROPERTIES',
+              updatedTypesQuery: Option.some({
+                bool: {
+                  should: updatedTypes.map((type) => ({ term: { type } })),
+                },
+              }),
+              logs: [
+                ...stateP.logs,
+                {
+                  level: 'info',
+                  message: `Kibana is performing a compatible update and NO root fields have been udpated. Kibana will update the following SO types so that ES can pickup the updated mappings: ${updatedTypes}.`,
+                },
+              ],
+            };
+          }
+        } else {
+          // some top-level properties have changed, e.g. 'dynamic' or '_meta' (see diffMappings())
+          return {
+            ...stateP,
+            controlState: 'UPDATE_TARGET_MAPPINGS_PROPERTIES',
```

Comment:
I wasn't sure about the `dynamic` case, so I played it safe and updated everything in that case.

Reply:
After some investigation, it seems that for _target_ indices we have `dynamic: 'strict'` and for temporary (and cloned) indices we have `dynamic: false`. 

So if a migrator fails before the `UPDATE_TARGET_MAPPINGS_PROPERTIES` step, the corresponding index would still have the `dynamic: false`. However it would be missing the `hashes` and the `indexTypesMap` as well, which already trigger a "full" pickup of documents.

TLDR: It can happen and we must pickup all documents in that case.